velox/functions/remote/client/Remote.cpp

```
@@ -53,6 +50,32 @@ class RemoteFunction : public exec::VectorFunction {
       serializedInputTypes_.emplace_back(serializeType(arg.type));
     }
     remoteInputType_ = ROW(std::move(types));
+    remoteClient_ = initializeClient();
+  }
+
+ private:
+  std::unique_ptr<RemoteClient> initializeClient() {
+    if (std::holds_alternative<folly::SocketAddress>(metadata_.location)) {
+      auto address = std::get<folly::SocketAddress>(metadata_.location);
+      return std::make_unique<ThriftRemoteClient>(
+          address,
+          functionName_,
+          remoteInputType_,
+          serializedInputTypes_,
+          metadata_);
+    }
+
+    if (std::holds_alternative<std::string>(metadata_.location)) {
+      auto url = std::get<std::string>(metadata_.location);
+      return std::make_unique<RestRemoteClient>(
+          url,
+          functionName_,
+          remoteInputType_,
+          serializedInputTypes_,
+          metadata_);
+    }
+
+    throw std::runtime_error("Unsupported client type in initializeClient");
```

please throw a velox exception instead using VELOX_UNSUPPORTED()